core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -375,6 +375,32 @@ copy_init:
 	bgt	copy_init
 #endif
 
+#ifdef CFG_TEE_CORE_PIE
+	/* Process relocations */
+
+	ldr	r0, =__rel_start
```

Comment:
How can this work? I'm expecting that these literals also will have relocation entries. Perhaps this only works if relocation doesn't update anything.

Reply:
No, because this file is not compiled `-fpic`, so that should be safe I think.

Reply:
I tried building this PR and I can find a `R_ARM_RELATIVE` entry for this literal.
You need to do something like I'm doing here:
https://github.com/jenswi-linaro/optee_os/blob/b12f2026525935e42fbe7aa367cf3bcaeea238d3/ldelf/start_a32.S#L16-L56

Reply:
Right, I can see the relocation now. But, why do we care? I'm assuming at this point the binary is loaded at the expected (physical) address. Then we do the relocs (possibly at a random VA for ASLR), and enable the MMU. Well that was the idea ;-)

Reply:
We care because it's hackish and can be easily avoided. :-)
I minor point is also, why imposing unnecessary restrictions on where a binary which can relocate itself is located.

Reply:
Yeah, I agree :-)